interceptor.go

```diff
@@ -25,12 +25,13 @@ type interceptor struct {
 
 // Interceptor is something that can intercept HTTP requests.
 type Interceptor interface {
-	Intercept(w http.ResponseWriter, req *http.Request, forwardInitialRequest bool, op ops.Op, defaultPort int)
+	Pipe(op ops.Op, w http.ResponseWriter, req *http.Request, defaultPort int, dial func(network, addr string) (net.Conn, error))
+	HTTP(op ops.Op, w http.ResponseWriter, req *http.Request, forwardInitialRequest bool, dial func(network, addr string) (net.Conn, error))
```

Comment:
Not new in this PR, but the `forwardInitialRequest` flag looks weird. I know it's hard to attach `X-Lantern-Persistent` with the first real request in `flashlight` due to current code structure, but it would be good if we can remove this flag later.